deb/openmediavault-bcache/usr/share/php/openmediavault/system/storage/bcache/cachedevice.inc

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ * This file is part of OpenMediaVault.
+ *
+ * @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
+ * @author    Michael Myrcik <michael.myrcik@web.de>
+ * @copyright Copyright (c) 2009-2024 Volker Theile
+ *
+ * OpenMediaVault is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * OpenMediaVault is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenMediaVault. If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace OMV\System\Storage\Bcache;
+
+/**
+ * Class to handle a bcache cache device.
+ */
+class CacheDevice extends BcacheDevice {
+	/**
+	 * Creates a backing device.
+	 * @return void
+	 * @throw \OMV\ExecException
+	 */
+	public function create() {
+		$cmdArgs = [];
+		$cmdArgs[] = "-C";
+		$cmdArgs[] = $this->getDeviceFile();
+		$cmd = new \OMV\System\Process("make-bcache", $cmdArgs);
+		$cmd->setRedirect2to1();
+		$cmd->execute();
+		while (!$this->isRegistered()) {
+			sleep(1);
+		}
+	}
+
+	/**
+	 * Get the state of cache device.
+	 * @return The state of cache device.
+	 * @throw \OMV\ExecException
+	 */
+	public function getState() {
+		if ($this->isRegistered()) {
+			return 1;
+		}
+		return 2;
+	}
+
+	/**
+	 * Shut down the cache set.
+	 * Waits until all attached backing devices have been shut down.
+	 * @return void
+	 * @throw \OMV\ExecException
+	 */
+	public function stop() {
+		while ($this->isRegistered()) {
+			$this->setSetValue("stop", 1);
```

Comment:
Is it really necessary to call this method on every iteration?

Reply:
I think once should be enough.